### ISSUE TITLE
Users created in wrong db

### ISSUE THREAD

**Issue by bsamsom** [Kaginari/terraform-provider-mongodb#37]
When looking at the examples:

There is an example of setting up the provider on the `admin` db:
https://github.com/Kaginari/terraform-provider-mongodb/blob/main/examples/main.tf#L18

but creating users under the `exemple` db
https://github.com/Kaginari/terraform-provider-mongodb/blob/main/examples/main.tf#L68

but when trying to do something similar, 

```
provider "mongodb" {
  ...
  auth_database = "admin"
}

resource "mongodb_db_user" "users" {
  auth_database = example
  name = test-user
  password = test-password
  role {
    role = "readWrite"
    db =   example
  }
}
```

and connecing to the db and running `db.getUsers()`
```
    {
      _id: 'test-user,
      user: 'test-user",
      db: 'admin',
      roles: [ { db: 'example', role: 'readWrite' } ]
    },
```

The db where the user is created is the `admin` db, instead of the expected `example` db

basically it seems to be missing a `use example` before the creation of the users.

best guess would be theres some confusion in the code of using the same name `auth_database` in both the provider and in the resource when they can be differnt values.

**Comment by ITMonta** [comment]
Hello @bsamsom  Thank you for your feed back can you specify which version of the provider you are using ?

**Comment by bsamsom** [comment]
I'm using version 0.1.7 of the provider with Document DB 5.0

**Comment by ITMonta** [comment]
Reviewed the current code closely for this: `resourceDatabaseUserCreate` reads the resource's own `auth_database` field (not the provider block's) into `database`, and `createUser(...)` runs MongoDB's `createUser` command via `client.Database(database).RunCommand(...)` — so the user is created against the resource's `auth_database`, which is what determines the user's home/auth db in MongoDB, independent of the provider-level connection's own `auth_database`. That's the correct behavior and matches what this issue expected.

The HCL snippet in the original report has unquoted values (`auth_database = example`, `name = test-user`) which isn't valid Terraform syntax, so it's possible that was a paste simplification that didn't reflect the exact config actually run. No reply came after the follow-up question here, and I can't reproduce the described behavior against the current codebase.

Closing as not reproducible against current code — please reopen with a fully reproducible config (matching `terraform validate`-clean HCL) if this is still happening on a recent version.
